fairlearn/metrics/metric_result.py

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+class MetricResult:
+    def __init__(self):
+        # The 'metric' property is the given metric evaluated without regard to group
+        self.metric = None
+        # The 'group_metric' array contains one entry for each group passed to the
+        # evaluation routine. The groups are required to be integers, starting from
```

Comment:
so is there a mapping from integers to groups? I'm surprised that we don't just use groups as defined by the user. What this describes could be expressed with a dictionary, too. Maybe I'm not seeing where the array comes in handy.

Reply:
@rihorn2 was quite keen on having the groups always be integers with a lower bound of zero. Otherwise JS-interop gets complicated with a dictionary.

Reply:
So we're making the user handle all that mapping back and forth?

Reply:
We would be requiring that mapping (although they _should_ be doing that anyway, or their data will be pointlessly large). We can discuss with @rihorn2

Reply:
Interesting. If that's the case we need to provide some really good examples of how to do that.

Reply:
Following comments from @MiroDudik , have gone back to the dictionary option which allows group_id to be of any type